pkg/platform/model/auth.go

```diff
@@ -0,0 +1,60 @@
+package model
+
+import (
+	"time"
+
+	"github.com/go-openapi/strfmt"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/pkg/platform/api/mono"
+	"github.com/ActiveState/cli/pkg/platform/api/mono/mono_client/oauth"
+	"github.com/ActiveState/cli/pkg/platform/api/mono/mono_models"
+)
+
+// RequestDeviceAuthorization posts a request to authorize this device on the Platform and
+// returns the device code needed for authorization.
+// The user is subsequently required to visit the device code's URI and click the "Authorize"
+// button.
+func RequestDeviceAuthorization() (*mono_models.DeviceCode, error) {
+	postParams := oauth.NewAuthDevicePostParams()
+	response, err := mono.Get().Oauth.AuthDevicePost(postParams)
+	if err != nil {
+		logging.Error("Error requesting device authorization: %v", err)
+		return nil, locale.NewError("err_auth_device")
```

Comment:
Use error wrapping instead. No need to log anything here, that'll happen down the chain when the error is "recognized".

Reply:
Okay

Reply:
This still isn't wrapping the error nor dropping the log.